### DISCUSSION TITLE
PLEASE HELP!!😥 Compiler Error: "Compiler run failed"

### DISCUSSION THREAD

**Discussion by jobmwitah** [Cyfrin/foundry-full-course-cu#133] [Q&A] [answered]
Am getting this error when I try to compile my project. 

```solidity
Error: 
Compiler run failed:
Error: Compiler error (/solidity/libsolidity/codegen/CompilerUtils.cpp:1430):Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.
```
I have spent more than 2 hours but I haven't found a solution please could someone help.🥺🙏🙏

**Comment by devblixt** [comment]
You might have declared too many local variables within a single scope. A good way to mitigate this would be to further modularize your code, like using custom getter functions. Plus, please drop your code.

**Reply by jobmwitah** [reply]
Is the same code like of Patrick's on lesson 9 "Lottery"

**Reply by jobmwitah** [reply, marked as answer]
Thanks It worked somehow. I removed some state variables I declared but I wasn't using on my Tests and It worked!!

**Reply by devblixt** [reply]
Yes, the error is because you inserted too many variables into a stack.

**Reply by jobmwitah** [reply]
Yeah I asked CHATGPT it told me to remove some local variables but all seem important.

**Comment by shedrick4570** [comment]
I started looking for variables to remove as well but then tried using the `--via-ir` tag as mentioned by @inallhonesty here: https://github.com/Cyfrin/foundry-smart-contract-lottery-f23/issues/10.

**Reply by sharonphiliplima** [reply]
What is your final solution for the `stack too deep` error?

**Reply by shedrick4570** [reply]
I had everything working fine up until adding the deployerKey, so I decided to just use the `--via-ir` tag to get around the issue. Try running `forge build --via-ir` and `forge test --via-ir`. It will take much longer to run, and IIRC, the tag would not work with certain other commands.

**Reply by sharonphiliplima** [reply]
Hey, I solved it by removing gasLane (adding comma and commenting it out)

**Comment by 0xriyadh** [comment]
This solves the issue
https://github.com/Cyfrin/foundry-smart-contract-lottery-f23/issues/9#issuecomment-1637113381
